### PR TITLE
Add OS.get_external_data_dir() to get Android external directory

### LINKED ISSUE #39414: Android: Read and write to app-specific external storage
**Godot version:**
3.2.1.rc1

**OS/device including version:**
Oculus Quest

**Issue description:**
Per [Android documentation](https://developer.android.com/training/data-storage), app can store data in external storage without requesting additional permissions as long as it's in the app-specific folder. On the Oculus Quest, this folder is `/sdcard/Android/data/com.yourpackage.name/`.

Now, this works if that folder already exists (e.g., created manually with adb). However, I can't create that folder through Godot using `make_dir()` or `make_dir_recursive(). It returns Error 20 (ERR_CANT_CREATE).

I would expect that either I can create the folder at that the folder is created automatically once read and write external storage permissions are grated in the Android Manifest.

I'm not sure if this is related to issue #38913 ?

**Steps to reproduce:**
- Enable read and write external storage in your Android export settings.
- Make sure your Android device (preferrable Oculus Quest) is attached to your computer
- Run this in a script's ready function:
```
if not dir.dir_exists("/sdcard/Android/data/com.yourpackage.name/"):
	var err = dir.make_dir_recursive("/sdcard/Android/data/com.yourpackage.name/")
	print("err ", err)


var open_err = dir.open("/sdcard/Android/data/com.yourpackage.name/")
print("open_err ", open_err)
```
- You'll get an error.
- Create that folder manually in adb:
```
adb mkdir "/sdcard/Android/data/com.yourpackage.name/"
```
- Run the script again - Success!

### PR DESCRIPTION
**Pull request by madmiraal**
Android provides each app with a dedicated location for external storage. However, this was often ignored because any external storage location could be accessed with the `READ_EXTERNAL_STORAGE` and `WRITE_EXTERNAL_STORAGE` permissions. With Android 10+, with [scoped storage](https://developer.android.com/training/data-storage#scoped-storage), the dedicated external storage location is the only external location that the app is allowed to access (without using the `MediaStore` API or requesting the `MANAGE_EXTERNAL_STORAGE` permission). On Android 10, this could be worked around with `android:requestLegacyExternalStorage="true"` (see #39103), however this workaround is disabled with Android 11 (see #47954).

This PR provides users with an easy way to locate the dedicated external storage location e.g.:
```
func _on_Save_Button_pressed():
	var dir = OS.get_external_data_dir()
	var image = $TextureRect.get_texture().get_data()
	image.save_png(dir + "/test.png")
```
To provide compatibility, on other OSs, this will return the same location as `OS.get_user_data_dir()`.

Fixes #39414.
Should help with #38913 and #48636.

### CODE REVIEW

File: platform/android/os_android.cpp [hunk 1]
```diff
@@ -222,6 +222,27 @@ String OS_Android::get_user_data_dir() const {
...
+		chdir(data_dir.utf8().get_data());
+		// Get the actual directory without the potential symlink.
+		char data_current_dir_name[2048];
+		getcwd(data_current_dir_name, 2048);
+		// Convert back to a String.
+		data_dir.parse_utf8(data_current_dir_name);
+		// Restore original current directory.
+		chdir(current_dir_name);
+		return data_dir;
+	}
```
Comment (m4gr3d) [inline platform/android/os_android.cpp:242]: This is the same logic as in `get_user_data_dir()`. Can you wrap the logic into a util function and have both functions call it instead.

**Review by m4gr3d — CHANGES_REQUESTED** [review]
Looks pretty good, thanks for adding the function! I just add one minor feedback to address, and it should be good for approval!

File: platform/android/os_android.cpp (hunk 1 above)
Comment (madmiraal) [reply platform/android/os_android.cpp]: Done.

**Review by m4gr3d — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by m4gr3d** [comment]
fyi @akien-mga @madmiraal I'm likely to revert this change and just make `get_user_data_dir()` behave as expected on Android 10 and higher.

**Comment by madmiraal** [comment]
> I'm likely to revert this change and just make get_user_data_dir() behave as expected on Android 10 and higher.

This is one of the many requirements to support Android's scoped storage. It should not be reverted. See my detailed [review](https://github.com/godotengine/godot/pull/50359#pullrequestreview-703581933) for more information.

**Pull request merged**